common/djangoapps/student/views.py

```diff
@@ -1384,7 +1384,9 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
             user = User.objects.get(email=email)
         except User.DoesNotExist:
             try:
-                user = User.objects.get(username=email)
+                # try using the email as username
+                username = email
```

Comment:
edX guys are very clever with their APIs, currently they do this 

```
from student.views import login_user
return shim_student_view(login_user, check_logged_in=True)(request)
``` 

on the ```user_api```. That is why I change it on the 

```common/djangoapps/student/views.py```